test/e2e_env/kubernetes/gateway/delegated/meshpassthrough.go

```diff
@@ -0,0 +1,113 @@
+package delegated
+
+import (
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"github.com/kumahq/kuma/pkg/plugins/policies/meshpassthrough/api/v1alpha1"
+	"github.com/kumahq/kuma/test/framework"
+	"github.com/kumahq/kuma/test/framework/client"
+	"github.com/kumahq/kuma/test/framework/envs/kubernetes"
+)
+
+func MeshPassthrough(config *Config) func() {
+	GinkgoHelper()
+
+	return func() {
+		AfterEach(func() {
+			Expect(framework.DeleteMeshResources(kubernetes.Cluster, config.Mesh, v1alpha1.MeshPassthroughResourceTypeDescriptor)).To(Succeed())
+		})
+
+		// we cannot test in a stable way first doing a successful request, because once connection is estabilished we don't break it after applying the policy (similar case to egress) the connection exists and works.
+		It("should control traffic to domains and IPs", func() {
+			esIP, err := framework.ServiceIP(kubernetes.Cluster, "external-service", config.NamespaceOutsideMesh)
+			Expect(err).ToNot(HaveOccurred())
+			// given
+			meshPassthrough := fmt.Sprintf(`
+apiVersion: kuma.io/v1alpha1 
+kind: MeshPassthrough
+metadata:
+  name: disable-passthrough
+  namespace: %s
+  labels:
+    kuma.io/mesh: %s
+spec:
+  targetRef:
+    kind: MeshSubset
+    proxyTypes: ["Gateway"]
+    tags:
+      kuma.io/service: delegated-gateway-admin_delegated-gateway_svc_8444
+  default:
+    passthroughMode: None
+`, config.CpNamespace, config.Mesh)
+
+			// when
+			err = kubernetes.Cluster.Install(framework.YamlK8s(meshPassthrough))
+
+			// we need to wait for a config to arrive because once request is done, connection is estabilished and it won't return 502
```

Comment:
Can we set a small idle timeout for connections, so we don't reuse it between requests?

Reply:
I think in this case we could try with `MeshProxyPatch` because it's on transparent proxy listener

Reply:
Changed to proxy patch, lets see if it's stable